baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/Slf4jLogsafeArgs.java

```diff
@@ -69,18 +69,14 @@ public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState
 
         List<? extends ExpressionTree> allArgs = tree.getArguments();
         int lastIndex = allArgs.size() - 1;
-        int startArg = ASTHelpers.isCastable(
-                ASTHelpers.getType(allArgs.get(0)),
-                state.getTypeFromString("org.slf4j.Marker"),
-                state) ? 2 : 1;
+        int startArg = MARKER.matches(allArgs.get(0), state) ? 2 : 1;
         ExpressionTree lastArg = allArgs.get(lastIndex);
         boolean lastArgIsThrowable = THROWABLE.matches(lastArg, state);
         int endArg = lastArgIsThrowable ? lastIndex - 1 : lastIndex;
 
         ImmutableList.Builder<Integer> badArgsBuilder = ImmutableList.builder();
-        Type argType = state.getTypeFromString("com.palantir.logsafe.Arg");
         for (int i = startArg; i <= endArg; i++) {
-            if (!ASTHelpers.isCastable(ASTHelpers.getType(allArgs.get(i)), argType, state)) {
```

Comment:
This appears to have missed a lot of cases that are correctly flagged using the new `ARG.matches` implementation.